GVRf/Extensions/3DCursor/IODevices/gvrmobileiodevice/src/main/java/com/example/gvrmobileiodevice/ConnectServer.java

```diff
@@ -0,0 +1,85 @@
+package com.example.gvrmobileiodevice;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.gearvrf.GVRContext;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+
+/**
+ * Created by v.nidamarthy on 8/29/16.
+ *
+ * Class to connect to the client, listens to the client, accepts to the connection
+ * and sets the input and output streams to read write the data
+ * The thread runs till the Server hears and accpets the client socket connection
+ * indicated by serverinfo.flag = true
+ * The class then spins of the read from server thread
+ */
+public class ConnectServer implements Runnable {
+    GVRMobileIoDevice.ServerInfo serverinfo = null;
+    String TAG = "gvrMobileDevice@ConnectServer";
```

Comment:
Actually this is not that good of a practice, in my opinion. It means you execute something on class load for no good reason. I find it wasteful.